server/src/completion.ts

```diff
@@ -191,13 +191,15 @@ function buildMatches(lookupName: string, elem: PerlElem, range: Range): Complet
     let documentation: MarkupContent | undefined = undefined;
     let docs: string[] = [];
 
-    if ( ["v", "c", "1"].includes(elem.type) && elem.typeDetail.length > 0) {
-        kind = CompletionItemKind.Variable;
-        detail = `${lookupName}: ${elem.typeDetail}`;
-    } else if ( ["v", "c", "1"].includes(elem.type) && lookupName == '$self' ) {
-        kind = CompletionItemKind.Variable;
-        // elem.package can be misleading if you use $self in two different packages in the same module. Get scoped matches will address this
-        detail = `${lookupName}: ${elem.package}`; 
+    if (["v", "c", "1"].includes(elem.type)) {
```

Comment:
This changes the behavior since not all "v" types match one of the two sub-conditions. "v" is also PerlSymbolKind.LocalVar which needs to be able to match later in the switch statement. These top two special cases are known variables types (either inferred from usage of `->new()` or because the variable is `$self`)